### DISCUSSION TITLE
[Admin] Comment se mettre admin et enregistrer son personnage ?

### DISCUSSION THREAD

**Discussion by mafiaa1997** [SOZ-Faut-etre-Sub/SOZ-FiveM-Server#11] [Q&A] [answered]
Bonsoir, 

Est-ce que quelqu'un sait comment se mettre les droits admin ? ca fonctionne pour le menu mapper mais admin aucune idée.
Et la 2ème question comment s'enregistrer correctement car après le skin du personnage créer et avoir noter oui, j'apparait à l'aéroport de los santos mais j'ai pas choisi mon nom et prénom. 

En vous remerciant d'avance pour votre aide

**Comment by joelwurtz** [comment, marked as answer]
> Est-ce que quelqu'un sait comment se mettre les droits admin ? ca fonctionne pour le menu mapper mais admin aucune idée.

Par défaut on est admin lorsque lance le serveur cf ce fichier qui permet d'attribuer un role par défaut pour les comptes non enregistrés: 

https://github.com/SOZ-Faut-etre-Sub/SOZ-FiveM-Server/blob/oss/global.cfg#L23

Tu peux check si tu as le droit admin en faisant pop une voiture (/car [model]) ou en appuyant sur F9 pour avoir le menu admin

> Et la 2ème question comment s'enregistrer correctement car après le skin du personnage créer et avoir noter oui, j'apparait à l'aéroport de los santos mais j'ai pas choisi mon nom et prénom.

Notre serveur utilise un système / bdd externe pour cela et cette partie n'est pas disponible en open source, cependant il suffit simplement de remplacer la fonction ici https://github.com/SOZ-Faut-etre-Sub/SOZ-FiveM-Server/blob/oss/resources/%5Bqb%5D/qb-core/server/functions.lua#L42

Avec le comportement suivant : 

 * renvoyer null si pas de compte
 * renvoyer un objet si il a un compte avec au moins les champs suivants : 
 
   * role : 'ADMIN' | 'USER' | 'HELPER' | 'STAFF'
   * name: prénom pour le personnage du joueur
   * lastName: nom de famille pour le personnage du joueur

**Comment by mafiaa1997** [comment]
Merci beaucoup je vais aller voir tout ça.

**Comment by lanahrone** [comment]
Hello @joelwurtz,

J'ai le même problème avec les droits admin qui sont manquants.

En revanche, si on `yarn dev` dans `soz-core`, on a bien la permission admin par défaut.

Suite au `yarn build`, les fichiers `build/server.js`, `build/client.js`, `build/nui.js` sont pourtant bien générés.

**Reply by joelwurtz** [reply]
Etrange pourtant cela ne change rien pour les droits admins, il y a peut être une histoire de première connection aussi

**Reply by AlexPresso** [reply]
J'avais le même souci, les droits admin sont bien présents mais sans `yarn dev`, c'est comme si les vues étaient manquantes, du coup rien ne s'affichait quand je faisais F9

**Reply by joelwurtz** [reply]
il doit y avoir un bug de compilation ou de chargement en mode build du coup, je regarderais si j'ai un peu de temps

**Reply by lanahrone** [reply]
@joelwurtz Pour info, c'est la même chose avec le téléphone.
Je ne vois pas le téléphone in-game sans faire de `yarn dev` dans `soz-phone`

**Reply by joelwurtz** [reply]
Faudrait voir pourquoi ca ne charge pas les scripts en mode build, j'ai pas ce problème la perso :/

**Reply by lanahrone** [reply]
@joelwurtz 
Autre info, si on a les convars `soz_core_hot_reload` et `soz_phone_hot_reload` à `false`, ça semble tout fonctionner avec les versions compilées de `soz-core` et `soz-phone`.

**Reply by joelwurtz** [reply]
alors du coup la c'est normal en mode true ca va cherche le js du nui sur le webpack dev server et en mode false ca va chercher le fichier, a voir comment mieux sync cela, pas forcement d'idée sur l'instant
